app/javascript/src/LoginPage/loginForms/Login3scaleForm.jsx

```diff
@@ -33,50 +39,40 @@ class Login3scaleForm extends React.Component<Props, State> {
     }
   }
 
-  setIsValidUsername = () => {
-    this.setState({isValidUsername: this.state.username !== ''})
-  }
-
-  handleTextInputUsername = (username: string) => {
-    this.setState({ username }, this.setIsValidUsername)
-  }
-
-  setIsValidPassword = () => {
-    this.setState({isValidPassword: this.state.password !== ''})
-  }
-
-  handleTextInputPassword = (password: string) => {
-    this.setState({ password }, this.setIsValidPassword)
+  handleInputChange = (value: string, event: SyntheticEvent<HTMLInputElement>) => {
+    const isValid = validateSingleField(event)
+    this.setState({
+      [namesToStateKeys[event.currentTarget.name].name]: value,
+      [namesToStateKeys[event.currentTarget.name].isValid]: isValid
+    })
   }
 
-  validateForm = (event: SyntheticEvent<HTMLInputElement>) => {
-    this.setIsValidUsername()
-    this.setIsValidPassword()
-    const isFormDisabled = this.state.username === '' ||
-      this.state.password === '' ||
-      this.state.isValidUsername === false ||
-      this.state.isValidPassword === false
-    if (isFormDisabled) {
+  validateForm = (event: SyntheticEvent<HTMLButtonElement>) => {
```

Comment:
I think this is unnecessary, we validate already on each input change.

Also, the submit button should be disabled whenever there's an error.

Reply:
But the user may press enter and submit the form, even by accident.

Reply:
What do you mean? Even if enter is pressed, the form will be validated already, there's no need to validate all fields again.

Reply:
I mean if the user clicks on enter without filling the form. The validation happens only when you fill the form